### PR TITLE
fix(consume,plugins): Fix suite duplication due to xdist, consume cli refactor

### PR DESCRIPTION
**Pull request by marioevz**
## 🗒️ Description
Fixes `test_suite` fixture from spawning duplicated test suites when xdist is used.

Reason being that xdist spawns a different process per worker to distribute the load, and the `session` or `module` scoped fixtures are not really shared between them.

Fix is to use the file system to create global locks per consume session, achieved by creating a temporary folder that is unique to each consume instance and is shared between the xdist processes, but do not disrupt other consume sessions.

The fixture also creates a lock per test-suite, so it is possible to run different test suites (i.e. one for `rlp` and one for `engine`) in the same consume instance.

Consume commands were also refactored to reduce code and `consume hive` command is now available which runs test suites of `consume rlp` and `consume engine` simultaneously.

## ✅ Checklist
- [x] All: Set appropriate labels for the changes.
- [x] All: Considered squashing commits to improve commit history.
- [x] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [x] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [x] Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been added to [converted-ethereum-tests.txt](/ethereum/execution-spec-tests/blob/main/converted-ethereum-tests.txt).
- [x] Tests: A PR with removal of converted JSON/YML tests from [ethereum/tests](/ethereum/tests) have been opened.
- [x] Tests: Included the type and version of evm t8n tool used to locally execute test cases:  e.g., ref with commit hash or geth 1.13.1-stable-3f40e65.
- [x] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

**Review by spencer-tb — APPROVED** [review]
Really nice! LGTM.

I'm going create a small PR based on this to split `pytest_commands.py  to the following within `src/cli`:
```tree
pytest_commands/
├── common.py
├── consume.py
├── fill.py
└── __init__.py
```
It should keep things clean when adding the fuzzer too.

Feel free to merge this or wait for my small refactor :D

File: src/cli/pytest_commands.py [hunk 1]
```diff
@@ -211,89 +192,109 @@ def consume():
...
+        if command_name == "all":
+            command_paths = consume_paths
+        elif command_name == "hive":
+            command_paths = consume_hive_paths
+        else:
+            test_path = consume_test_path(command_name, hive)
+            consume_paths.append(test_path)
+            command_paths = [test_path]
+            if hive:
+                consume_hive_paths.append(test_path)
```
[the remark this answers is not in the document]
Comment (spencer-tb) [reply src/cli/pytest_commands.py:226]: Added some changes here. Let me know what you think, maybe I missed something but I don't think we need the consume paths. Only the command paths.

https://github.com/ethereum/execution-spec-tests/pull/714

**Comment by spencer-tb** [comment]
> The fixture also creates a lock per test-suite, so it is possible to run different test suites (i.e. one for rlp and one for engine) in the same consume instance.

Just so I understand this essentially means running `consume hive` with parallelism where only one instance is spawned for both `rlp` and `engine`?

**Comment by marioevz** [comment]
> > The fixture also creates a lock per test-suite, so it is possible to run different test suites (i.e. one for rlp and one for engine) in the same consume instance.
> 
> 
> 
> Just so I understand this essentially means running `consume hive` with parallelism where only one instance is spawned for both `rlp` and `engine`?
> 
> 

You can either launch `consume rlp` and `consume engine` separately or do `consume hive` with any xdist number parameter, and the result in hive view should be exactly the same.

Basically we could reduce the code in the hive cluster script.

**Pull request merged**